pkg/provider/remote/gcp.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remote
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/AlecAivazis/survey/v2"
+	"gopkg.in/yaml.v2"
+)
+
+// TODO: Move this into remote provider logic
+// A provider should be able to produce it's own template for a valid stack specification as stack specifications may change over time
+type gcpProvider struct {
+	*nitricDeployment
+}
```

Comment:
These are just bridging providers while we shift this functionality into the remote providers themselves, we can probably also inject additional bridging logic into them as well, for migrating breaking changes from older provider definitions.